apis/v1alpha1/httproute_types.go

```diff
@@ -285,12 +295,38 @@ type ForwardToTarget struct {
 	// Support: Core
 	//
 	// +optional
-	TargetPort *TargetPort `json:"targetPort" protobuf:"bytes,2,opt,name=targetPort"`
+	TargetPort *TargetPort `json:"targetPort,omitempty" protobuf:"bytes,2,opt,name=targetPort"`
+
+	// Weight specifies the proportion of traffic to be forwarded to a targetRef,
+	// computed as weight/(sum of all weights in targetRefs). The following example
+	// sends 70% of traffic to service "my-trafficsplit-sv1" and 30% of the traffic
+	// to service "my-trafficsplit-sv2":
+	//
+	//   forwardTo:
+	//     - targetRef:
+	//         name: my-trafficsplit-sv1
+	//         weight: 70
+	//     - targetRef:
+	//         name: my-trafficsplit-sv2
+	//         weight: 30
+	//
+	// If only one targetRef is specified, weight is ignored and all traffic is
+	// forwarded to the targetRef. If multiple targetRefs are specified without
+	// weight, traffic is evenly distributed across all targetRefs.
```

Comment:
Consider making weight required and using CRD defaults to set a default value. We can simplify by e.g. using weight: 1 when not specified. This makes behavior intuitive when a weight isn't supplied, and makes weighting functionality more discoverable as users will see a value to play with when viewing their configuration.

Reply:
@JeremyOT commit `9abadaa` updates `weight` based on your feedback above. Thanks for the review.